### PR TITLE
Remove esbuild@0.11.x from dependency list

### PR DESCRIPTION
**Pull request by nayaabkhan**

### CODE REVIEW

File: packages/@contentlayer/core/package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   },
   "peerDependencies": {
     "date-fns": "2.x",
-    "esbuild": "0.11.x || 0.12.x || 0.13.x",
+    "esbuild": "^0.12.1 || 0.13.x",
```
Comment (nayaabkhan) [inline packages/@contentlayer/core/package.json:17]: I've not tested if this way of specifying the versions works.

File: packages/@contentlayer/core/package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "camel-case": "^4.1.2",
     "comment-json": "^4.2.2",
     "date-fns": "2.x",
-    "esbuild": "0.11.x || 0.12.x || 0.13.x || 0.14.x",
+    "esbuild": "^0.12.1 || 0.13.x || 0.14.x",
```
Comment (nayaabkhan) [inline packages/@contentlayer/core/package.json:36]: I've not tested if this way of specifying the versions works.

**Comment by schickling** [comment]
Could you please provide some explanation/motivation for this change? Which problem are you trying to address with this? (Please refer to the corresponding issues.)

**Comment by nayaabkhan** [comment]
Reproduction repo: https://github.com/contentlayerdev/contentlayer/issues/205#issuecomment-1114147536

**Comment by schickling** [comment]
Thanks a lot for this PR and corresponding issue! Very helpful! 🙏 

Will be released as part of the upcoming `0.2.5`.

**Comment by nayaabkhan** [comment]
No problemo! Very happy to help out.

**Pull request merged**